queries/lua/indents.scm

```diff
@@ -1,31 +1,16 @@
 [
   (function_definition)
   (variable_declaration)
-  (local_variable_declaration)
   (field)
-  (local_function)
-  (function)
+  (do_statement)
+  (while_statement)
+  (repeat_statement)
   (if_statement)
   (for_statement)
-  (for_in_statement)
-  (repeat_statement)
   (return_statement)
-  (while_statement)
-  (table)
+  (table_constructor)
   (arguments)
-  (do_statement)
+  (block)
 ] @indent
 
-[
-  "end"
-  "until"
-  "{"
-  "}"
-  "("
-  ")"
-  "then"
-  (else)
-  (elseif)
-] @branch
```

Comment:
Can we add back these removed bits? That might fix the tests. This is the equivalent:

```scheme
(else_statement) @branch
(elseif_statement) @branch

[
  "end"
  "until"
  "{"
  "}"
  "("
  ")"
  "then"
] @branch
```

Reply:
Yep, sounds good, just did. We might iterate a bit to get the best of both worlds in all queries.

Reply:
Still some failures, but we're getting there ;)

Reply:
I'm setting up tests locally and looking into the indents test failure.

Reply:
OK, I think the main culprit is the `block` capture; we can't use that for indentation since it's used both where you want to and where you don't want to indent.

Reply:
There's only one test failure remaining (`loop.lua`), which I can't reproduce locally. In fact, the output looks weird: the mis-indented line isn't even in the test file?

Reply:
One failure remaining (`loop.lua`); this indents correctly when the whole file (with the new line) is indented via `gg=G`, but inserting a new line adds a wrong indent. Not sure what is happening there.

Reply:
I've added the `break_statement` to `@ignore` -- @theHamsta is that a reasonable thing to do?

Reply:
But locally I then get wrongly indented newlines in `for` loops...

Reply:
This is not related to `break_statement`.

So for 

```scheme
while false do
  a = 1
  b = 2
end
```

and ast

```
while_statement [93, 0] - [96, 3]
  condition: false [93, 6] - [93, 11]
  body: block [94, 2] - [95, 7]
    assignment_statement [94, 2] - [94, 7]
      variable_list [94, 2] - [94, 3]
        name: identifier [94, 2] - [94, 3]
      expression_list [94, 6] - [94, 7]
        value: number [94, 6] - [94, 7]
    assignment_statement [95, 2] - [95, 7]
      variable_list [95, 2] - [95, 3]
        name: identifier [95, 2] - [95, 3]
      expression_list [95, 6] - [95, 7]
        value: number [95, 6] - [95, 7]
```

i.e. if there are more than one statements in `body:` and cursor is on the first statement, if newline is entered,

https://github.com/nvim-treesitter/nvim-treesitter/blob/4cf688ee681bbe9a35651732b2d2281a01c40066/lua/nvim-treesitter/indent.lua#L77

is getting `body: block`, instead of getting the first statement inside `block`.

Reply:
OK, that looks like something that needs to be fixed in general. (If you want to take a stab at it, that would be very welcome ;))

I've marked this test as expected failure for now.

Reply:
@theHamsta can you see why the tests are still failing, even though I marked the tests as `XFAIL`?

Reply:
> can you see why the tests are still failing, even though I marked the tests as XFAIL or added them to the expected_failures table?

I think these 2 needs to be removed from `expected_failures`.

![image](https://user-images.githubusercontent.com/8050659/149994503-49065bf0-8c5f-462e-bda8-70a52595d3b2.png)

Reply:
Oh, you mean they are failing because the expected failures _don't_ fail now? That's... not intuitive.

Reply:
Indeed, that was it, good catch!